### ISSUE TITLE
no Java certs

### ISSUE THREAD

**Issue by themaddoctor** [lfs-book/make-ca#6]
I installed OpenJDK 12.0.2+10, then I installed make-ca 1.5, then I ran
make-ca -g
as root, and it did a lot of things to my /etc/pki directory, but there are no Java certificates.

keytool -list -cacerts
Enter keystore password:

*****************  WARNING WARNING WARNING  *****************
* The integrity of the information stored in your keystore  *
* has NOT been verified!  In order to verify its integrity, *
* you must provide your keystore password.                  *
*****************  WARNING WARNING WARNING  *****************

Keystore type: JKS
Keystore provider: SUN

Your keystore contains 0 entries

**Comment by themaddoctor** [comment]
```
make-ca -g --force
...


Certificate:  Starfield Services Root Certificate Authority - G2
Keyhash:      09789157
Added to p11-kit anchor directory with trust 'C,,'.


/
Extracting OpenSSL certificates to /etc/ssl/certs...Done!
Extracting GNUTLS server auth certificates to /etc/pki/tls/certs/ca-bundle.crt...Done!
Extracting GNUTLS S-Mime certificates to /etc/pki/tls/certs/email-ca-bundle.crt...Done!
Extracting GNUTLS code signing certificates to /etc/pki/tls/certs/objsign-ca-bundle.crt...Done!
Extracting Java cacerts (JKS) to /etc/pki/tls/java/cacerts...Done!
```

But the resulting /etc/pki/tls/java/cacerts has only 32 bytes and no certificates.

**Comment by themaddoctor** [comment]
I have just noticed that
/etc/pki/tls/certs/ca-bundle.crt
/etc/pki/tls/certs/objsign-ca-bundle.crt
/etc/pki/tls/certs/email-ca-bundle.crt
all have zero size.

**Comment by djlucas** [comment]
This implies that the download failed. Best guess is that /etc/resolv.conf was not populated. Can you confirm?

**Comment by themaddoctor** [comment]
No. All my browsers are finding webpages. /etc/resolv.conf contains:
nameserver 8.8.8.8

**Comment by djlucas** [comment]
And OpenSSL is installed already?

**Comment by themaddoctor** [comment]
yes

**Comment by djlucas** [comment]
Okay, so is /etec/ssl/certdata.txt empty as well? How about /etc/ssl/certs/* or /etc/pki/anchors/*? Is p11-kit installed? I'll obviously need to guard against any missing BLFS tools that are required for the individual steps.

**Comment by themaddoctor** [comment]
-rw-r--r-- 1 root root 1269904 Jan 11 13:59 /etc/ssl/certdata.txt
/etc/ssl/certs/ is an empty directory; not sure why
/etc/pki/anchors contains about 150 .pem files
yes, p11-kit is installed (0.20.2)
thank you

**Comment by djlucas** [comment]
That's a really old version of P11-Kit, which I suspect is the issue. If built using BLFS instruction for 0.20, it is probably trying to use /etc/ssl/certs as the anchor directory. It also probably doesn't know how to handle the Mozilla Trust extensions. Is it possible to update to 0.23 at least?

**Comment by themaddoctor** [comment]
Hold on... I'll try that in the morning.

**Comment by djlucas** [comment]
Is this resolved with the new P11-Kit and the correct anchors directory?

**Comment by themaddoctor** [comment]
Ouch! I went on vacation and completely forgot. Will compile the new p11-kit and give you a real answer.
Thanks for being patient.

**Comment by themaddoctor** [comment]
... Everything looks good, with one exception.
/etc/pki/tls/certs/objsign-ca-bundle.crt is empty. Is this normal?

**Comment by djlucas** [comment]
Yes, this is normal. Mozilla does not care about signing certificates, but it is entirely possible that you'll need them for your own software projects. Thanks for getting back to me on this. I'll add some additional notes in the README about software versions and paths.

**Comment by themaddoctor** [comment]
Thank YOU.

**Comment by djlucas** [comment]
Fixed in #81786ef.
